oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java

```diff
@@ -202,6 +206,13 @@ private static void mapIndexRules(ElasticIndexDefinition indexDefinition, XConte
                 }
             }
             mappingBuilder.endObject();
+
+            if (useInSimilarity) {
```

Comment:
sim:, dense_vector can be moved to some constant. And dims value = 1024 should be made configurable because this can change based on the underlying implementation by dam (which is not in our control)

Reply:
Done.